docs/src/structuring_element.md

```diff
@@ -0,0 +1,197 @@
+```@setup concept_se
+using ImageMorphology
+using ImageBase
+using TestImages
+```
+
+# Structuring element
+
+Structuring Element (SE) is the key concept in morphology to indicate the connectivity and the
+neighborhood. This page explains this structuring element concept, and how ImageMorphology supports
+the general SEs without compromising the performance on the most commonly used special SE cases.
+
+## The erosion example
+
+The erosion `erode` function in its simplest 1-dimensional case can be defined as
+
+$$\varepsilon_A[p] = min(A[p-1], A[p], A[p+1])$$
+
+Because the output value at position $p$ not only depends on its own pixel `A[p]` but also on
+its neighborhood values `A[p-1]` and `A[p+1]`, we call this type of operation a _neighborhood
+image transformation_.
+
+Now comes the question: **if we try to generalize the `erode` function, what should we do?** --
+we would like to generalize the concept of "neighborhood".
+
+## Two neighborhood representations
+
+By saying "$\Omega_p$ is the neighborhodd of $p$", we are expressing `p in Ωₚ` in plain Julia. For
+performance consideration, this `Ωₚ` is usually generated from the `(p, Ω)` pair; we call this
+template `Ω` a _structuring element_. There are two ways to expressing this `Ω`:
+
+- **displacement offset**: a list of `CartesianIndex` to inidcate the offset to the center point `p`
+- **connectivity mask**: a bool array mask to indicate the connectivity to the center point `p`
+
+For instance, in the following code block we build a commonly named C4 connectivity in the 2-dimensional case:
+
+```@example concept_se
+# displacement offset
+Ω_offsets = [
+    CartesianIndex(-1, 0),
+    CartesianIndex(0, -1),
+    CartesianIndex(0, 1),
+    CartesianIndex(1, 0),
+]
+
+# connectivity mask
+Ω_bool = Bool[
+    0 1 0
+    1 1 1
+    0 1 0
+]
+nothing #hide
```

Comment:
For conceptual consistency perhaps we should consider making this have axes `(-1:1, -1:1)`.